test/protocol/insurance.js

```diff
@@ -0,0 +1,271 @@
+"use strict";
+
+const assertEx = require("../testutils.js");
+
+const Amber = artifacts.require("./Amber.sol");
+const Market = artifacts.require("./protocol/Market/Market.sol");
+const Offer = artifacts.require("./protocol/Market/Offer.sol");
+const MeasurementsOnChain = artifacts.require("./protocol/Measurements/MeasurementsOnChain.sol");
+const RangeRequirementsFactory = artifacts.require("./protocol/Requirements/RangeRequirementsFactory.sol");
+const RangeValidator = artifacts.require("./protocol/Validator/RangeValidator.sol");
+const EscrowedAgreement = artifacts.require("./protocol/Agreement/EscrowedAgreement.sol");
+const Insurance = artifacts.require("./protocol/Insurance/Insurance.sol");
+
+contract('Insurance', function(accounts) {
+    var amber;
+    var agreement;
+    var insurance;
+    
+    var market, offer, measurements, requirements, validator;
+    var buyer, seller, amberOwner, marketOwner, insuranceOwner;
+
+    var newAgreementInsuredListener;
+    var reimbursementRequestedListener;
+
+    beforeEach("deploy contracts", async() => {
+        buyer = accounts[1];
+        seller = accounts[2];
+        amberOwner = accounts[9];
+        marketOwner = accounts[8];
+        insuranceOwner = accounts[7];
+
+        let result = await web3.eth.getBlock('earliest');
+        let startTime = result.timestamp;
+        amber = await Amber.new(startTime, startTime, {from: amberOwner});
```

Comment:
I would consider using MockToken:
https://github.com/OpenZeppelin/zeppelin-solidity/blob/5751eac44401f44ca9d27d396341ea7ac1f73f46/contracts/mocks/BasicTokenMock.sol

Reply:
As discussed over Skype, the implementation of Amber in Ambrosus branch is like a mock ERC20 token that allows minting tokens without re-creating the contract which is very handy in this case because otherwise would need to recreate entire test env (Market, Offer, Insurance, etc.) in each test defeating the purpose of common beforeEach function.

Therefore I prefer to keep using Amber in these tests.